sktime/transformations/panel/augmenter.py

```diff
@@ -0,0 +1,1217 @@
+        if y is not None:
+            y_aug = y.iloc[idx_list]
+            y_aug.index = X_aug.index
+            return X_aug, y_aug, idx_list
+        else:
+            return X_aug, idx_list
+
+    def _plot_augmentation_example(self, X, y):
+        """Plot original and augmented instance examples for each variable.
+
+        This is a wrapper function calling the static function
+        plot_augmentation_example() for compatibility reasons.
+        """
+        plot_augmentation_example(self, X, y)
+
+
+# static functions
+def plot_augmentation_example(
+    fitted_transformer,
+    X,
+    y=None,
+    n_instances_per_variable=5,
+):
+    """Plot original and augmented instance examples for each variable.
+
+    Parameters
+    ----------
+    fitted_transformer: fitted transformer
+        A fitted (augmentation) transformer.
+    X: Panel of pd.DataFrame
+        Uni- or multivariate dataset.
+    y: Series or Panel, optional (default=None)
+        Target variable, if y is available and of categorical scale,
+        it will be used to stratify the randomly drawn examples.
+    n_instances_per_variable: int, optional (default = 5)
+        number of time series to draw per variable (row).
+
+    Returns
+    -------
+    matplotlib.figure.Figure: A figure with a [n_variables, 2] subplot-grid.
+    """
+    _check_soft_dependencies("matplotlib")
+    import matplotlib.pyplot as plt
+
+    n_vars = X.shape[1]  # get number of variables of X
+    # pick (stratified regarding categorical y) examples from the original input
+    # data
+    X, y, idx = SeqAugPipeline.draw_random_samples(
+        X,
+        y,
+        n=n_instances_per_variable,
+        shuffle_and_stratify=True,
+        without_replacement=True,
+    )
+    # make sure, that the given transformer is fitted
+    if not fitted_transformer._is_fitted:
+        fitted_transformer.fit(X, y)
+    # get augmented data
+    Xt = fitted_transformer.transform(X, y)
+
+    # create description string (parameterization of the augmenter)
+    try:
+        ft = fitted_transformer
+        if isinstance(ft._param, (float, int)):
+            help_param = f"{ft._param:.4f}"
+        else:
+            help_param = str(ft._param)
+        nl = "\n"
+        param_str = (
+            f"{ft.__class__.__name__} with parameters{nl}"
+            f"p={ft.p:.2f}, "
+            f"param={help_param}, "
+            f"use_relative_fit={ft.use_relative_fit},{nl}"
+            f"relative_fit_stat_fun={ft.relative_fit_stat_fun.__name__}, "
+            f"relative_fit_type='{ft.relative_fit_type}', "
+            f"random_state={ft.random_state},{nl}"
+            f"excluded_var_indices={ft.excluded_var_indices}, "
+            f"n_jobs={ft.n_jobs}.{nl}"
+            f"Date: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
+        )
+    except Exception as e:
+        # param_str = "Unknown augmenter and parameterization"
+        raise ModuleNotFoundError(e)
```

Comment:
Any reason to call `ModuleNotFoundError` here?

Reply:
Not really, I fixed that. Notice that `plot_augmentation_example()` is a nice feature for the developement of augmented TS but not mandatory as it provides no essential functionality. Maybe we will move this to the example notebook later.